api_app/analyzers_manager/observable_analyzers/criminalip.py

```diff
@@ -0,0 +1,91 @@
+from typing import Dict
+
+import requests
+
+from api_app.analyzers_manager import classes
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+
+class CriminalIp(classes.ObservableAnalyzer):
+    url = "https://api.criminalip.io"
+    _api_key: str = None
+
+    malicious_info: bool = True
+    privacy_threat: bool = False
+    is_safe_dns_server: bool = False
+    suspicious_info: bool = False
+    banner_search: bool = True
+    banner_stats: bool = False
+
+    def update(self):
+        pass
+
+    def make_request(self, url: str, params: Dict[str, str] = None) -> Dict:
+        headers = {"x-api-key": self._api_key}
+        r = requests.get(url, headers=headers, params=params)
+        r.raise_for_status()
+        return r.json()
+
+    def run(self):
+        URLs = {
+            self.ObservableTypes.IP.value: {
+                "endpoints": {
+                    "malicious_info": "/v1/feature/ip/malicious-info",
+                    "privacy_threat": "/v1/feature/ip/privacy-threat",
+                    "is_safe_dns_server": "/v1/feature/ip/is-safe-dns-server",
+                    "suspicious_info": "/v2/feature/ip/suspicious-info",
+                },
+                "params": {"ip": self.observable_name},
+            },
+            self.ObservableTypes.DOMAIN.value: {
+                "endpoints": {
+                    "default": "/v1/domain/quick/hash/view",
```

Comment:
does this work? there's no attribute `domain`, right?

I mean, it's cool to try to write python code which is more structured, with less lines of code, and seems super cool.

However, in my experience sometimes it is less useful as expected, leads to possible hidden bugs and it is more complicated to maintain. I am not asking to change this now but be careful

Reply:
ah crap! You're right. I'll push a fix. It does get tricky.